core/json/src/main/scala/net/liftweb/json/JsonParser.scala

```diff
@@ -83,89 +85,118 @@ object JsonParser {
       case e: Exception => throw new ParseException("parsing failed", e)
     } finally { buf.release }
   }
-  
-  private[json] def unquote(string: String): String = 
+
+  private[this] final val HexChars: Array[Int] = {
+    val chars = new Array[Int](128)
+    var i = 0
+    while (i < 10) {
+      chars(i + '0') = i
+      i += 1
+    }
+    i = 0
+    while (i < 16) {
+      chars(i + 'a') = 10 + i
+      chars(i + 'A') = 10 + i
+      i += 1
+    }
+    chars
+  }
+
+  private[json] def unquote(string: String): String =
     unquote(new JsonParser.Buffer(new java.io.StringReader(string), false))
-  
-  private[json] def unquote(buf: JsonParser.Buffer): String = {
-    def unquote0(buf: JsonParser.Buffer, base: String): String = {
-      val s = new java.lang.StringBuilder(base)
+
+  private[this] def unquote(buf: JsonParser.Buffer): String = {
+    def unquote0(buf: JsonParser.Buffer): String = {
+      val builder = buf.builder
+      builder.delete(0, builder.length())
       var c = '\\'
       while (c != '"') {
         if (c == '\\') {
-          buf.next match {
-            case '"'  => s.append('"')
-            case '\\' => s.append('\\')
-            case '/'  => s.append('/')
-            case 'b'  => s.append('\b')
-            case 'f'  => s.append('\f')
-            case 'n'  => s.append('\n')
-            case 'r'  => s.append('\r')
-            case 't'  => s.append('\t')
+          buf.substring(intoBuilder = true)
+          (buf.next: @switch) match {
+            case '"'  => builder.append('"')
+            case '\\' => builder.append('\\')
+            case '/'  => builder.append('/')
+            case 'b'  => builder.append('\b')
+            case 'f'  => builder.append('\f')
+            case 'n'  => builder.append('\n')
+            case 'r'  => builder.append('\r')
+            case 't'  => builder.append('\t')
             case 'u' =>
+              var byte = 0
+              var finalChar = 0
               val chars = Array(buf.next, buf.next, buf.next, buf.next)
-              val codePoint = Integer.parseInt(new String(chars), 16)
-              s.appendCodePoint(codePoint)
-            case _ => s.append('\\')
+              while (byte < 4) {
+                finalChar = (finalChar << 4) | HexChars(chars(byte).toInt)
```

Comment:
The `chars` array contains the bytes that you want in the 64-bit `finalChar` in the “correct” placement. So it goes something like this if the values in `chars` are 1234:

```
finalChar     chars (> is current index)
  0000        Array(1,4,2,3)
  0001        Array(>1,4,2,3)
  0014        Array(1,>4,2,3)
  0142        Array(1,4,>2,3)
  1423        Array(1,4,2,>3)
```